### PR TITLE
Add Environment class and Environment_TEST.

### PR DESCRIPTION
**Pull request by LucasJSch**
This PR adds the class implementation of `Environment`, together with the unit tests for that class.

Additionally, some other minor changes were made to the `Insect`,  `InsectMetrics` and `Cell` classes to fix minor bugs.

Some `CMakeLists` files were changed to compile the new unit tests and to add the new `environment.cc` file into the source files macro.

### CODE REVIEW

File: course/src/environment.cc [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    }
+    for (const insect_ptr& insect : insects) {
+        insect->RunRound();
+    }
+    // Delete the dead insects from the list.
+    insects.erase(std::remove_if(insects.begin(), insects.end(),
+                                 [](insect_ptr& insect){
+                                     return insect->IsDead();
+                                    }),
+                                 insects.end());
```
Comment (ernestmc) [inline course/src/environment.cc:111]: :+1:

File: course/test/src/Environment_TEST.cc [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+GTEST_TEST(EnvironmentTest, SimulationNotFinishedWhenCreatedInsects) {
+    Environment env(2, 2, 1, 1);
+    EXPECT_FALSE(env.SimulationFinished());
+}
+
+GTEST_TEST(EnvironmentTest, SimulationFinishedWhenThereIsASingleCellAndASingleInsect) {
+    Environment env1(1, 1, 0, 1);
+    EXPECT_TRUE(env1.SimulationFinished());
+    Environment env2(1, 1, 0, 1);
```
Comment (ernestmc) [inline course/test/src/Environment_TEST.cc:50]: This is the same case as above

**Review by ernestmc — APPROVED** [review]
lgtm

File: course/test/src/Environment_TEST.cc (hunk 2 above)
Comment (LucasJSch) [reply course/test/src/Environment_TEST.cc:50]: You're right. I just wanted to make sure that the condition applies both when you start with only an Ant and only a Doodlebug.

**Pull request merged**